R/tm_a_pca.R

```diff
@@ -312,86 +304,77 @@ srv_a_pca <- function(id, datasets, reporter, dat, plot_height, plot_width, ggpl
         )
       ))
 
-      teal.code::chunks_push(
-        id = "keep_columns_call",
-        expression = substitute(
-          expr = {
-            keep_columns <- keep_cols
-          },
-          env = list(
-            keep_cols = keep_cols
-          )
+      quosure <- teal.code::eval_code(
+        merged$anl_q_r(),
+        substitute(
+          expr = keep_columns <- keep_cols,
+          env = list(keep_cols = keep_cols)
         ),
-        chunks = chunks_stack
+        name = "keep_columns_call"
       )
 
       if (na_action == "drop") {
-        teal.code::chunks_push(
-          id = "ANL_drop_na_call",
-          expression = substitute(
-            expr = ANL <- tidyr::drop_na(ANL, keep_columns), # nolint
-            env = list()
-          ),
-          chunks = chunks_stack
+        quosure <- teal.code::eval_code(
+          quosure,
+          quote(ANL <- tidyr::drop_na(ANL, keep_columns)), # nolint
+          name = "ANL_drop_na_call"
         )
       }
 
       if (scale) {
-        teal.code::chunks_validate_custom(
-          substitute(
-            expr = vapply(ANL[keep_cols], function(column) length(unique(column)) != 1, FUN.VALUE = logical(1)),
-            env = list(keep_cols = keep_cols)
-          ),
-          msg = paste0(
-            "You have selected `Center & Scale` under `Standardization` in the `Pre-processing` panel, ",
-            "but one or more of your columns has/have a variance value of zero, indicating all values are identical"
-          ),
-          chunks = chunks_stack
+        code <- substitute(
+          vapply(ANL[keep_cols], function(column) length(unique(column)) != 1, FUN.VALUE = logical(1)),
+          env = list(
+            keep_cols = keep_cols
+          )
         )
+        msg <- paste0(
+          "You have selected `Center & Scale` under `Standardization` in the `Pre-processing` panel, ",
+          "but one or more of your columns has/have a variance value of zero, indicating all values are identical"
+        )
+        validate(need(all(eval(code, quosure@env)), msg))
       }
```

Comment:
why does this need to be substituted and evaluated?
Can we just evaluate this directly and validate?

Reply:
fixed